admin_guide/ipsec.adoc

```diff
@@ -0,0 +1,174 @@
+These certificates can also be used to secure pod communications with IPsec. You
+need at three files on each host:
+
+- cluster Certificate Authority file
+- host client certificate file
+- host private key file
+
+First, determine what the certificate's nickname will be after it has been
+imported into the `libreswan` certificate database. The nickname is taken
+directly from the certificate's subject's Common Name (CN):
+
+----
+openssl x509 -in /path/to/client-certificate -subject -noout | sed -n 's/.*CN=\(.*\)/\1/p'
+----
+
+Save the nickname for later.
+
+The client certificate, CA certificate, and private key files must be combined
+into a PKCS#12 file, which is a common container format for multiple
+certificates and keys.  To do this, you can use the `openssl` program:
+
+----
+  openssl pkcs12 -export \
+    -in /path/to/client-certificate \
+    -inkey /path/to/private-key \
+    -certfile /path/to/certificate-authority \
+    -passout pass: \
+    -out certs.p12
+----
+
+This PKCS#12 file must then be imported into the `libreswan` certificate
+database. The -W option is left empty because we do not need to assign a
+password to the PKCS#12 file as it is only temporary.
+
+----
+  ipsec initnss
+  pk12util -i certs.p12 -d sql:/etc/ipsec.d -W ""
+----
+
+[[ipsec-policy]]
+=== libreswan IPsec Policy
+
+Now that the necessary certificates have been imported into the `libreswan`
+certificate database, you can create a policy that uses them to secure
+communication between hosts in your cluster. The following configuration
+creates two `libreswan` connections.  The first encrypts traffic using the
+Openshift certificates, while the second creates exceptions to the encryption
+for cluster-external traffic.
+
+Place the following text into the file /etc/ipsec.d/openshift-cluster.conf,
+being sure to replace the `<certificate-nickname-here>` text with the 
+certificate nickname you saved above:
+
+----
+conn private
+	left=%defaultroute
+	leftid=%fromcert
+	# our certificate
+	leftcert="NSS Certificate DB:<certificate-nickname-here>"
+	right=%opportunisticgroup
+	rightid=%fromcert
+	# their certificate transmitted via IKE
+	rightca=%same
+	ikev2=insist
+	authby=rsasig
+	failureshunt=drop
+	negotiationshunt=hold
+	auto=ondemand
+
+conn clear
+	left=%defaultroute
+	right=%group
+	authby=never
+	type=passthrough
+	auto=route
+	priority=100
+----
+
+Now that the configuration has been defined, you need to tell `libreswan`
+what IP subnets and hosts to apply each policy to. This is done through policy
+files in /etc/ipsec.d/policies/ where each configured connection has a
+corresponding policy file. In our example above, we have two connections,
+`private` and `clear` and each will have a file in /etc/ipsec.d/policies.
+
+/etc/ipsec.d/policies/private should contain the IP subnet of your cluster,
+which your hosts receive IP addresses from. This, by default, will cause all
+communication between hosts in the 172.16.0.0/16 subnet to be encrypted if
```

Comment:
Makes it sound like regardless of what you put in the file, it will affect traffic to 172.16.0.0/16... I think you rearranged some text here and didn't fully fix up everything for it.